### PR TITLE
Implemented GEOMEAN function

### PR DESCRIPTION
**Pull request by gallonfizik**
Tested against Excel 2007 and according to [MS Excel docs](https://support.office.com/en-us/article/geomean-function-db1ac48d-25a5-40a0-ab83-0b38980e40d5)
All POI tests pass.
GeometricMean behavior for series containing zeros is handled.
There's also a change to MultiOperandNumeric function: now it handles MissingArgEvals (counts as 0).

It will make sense to squash the branch.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: src/java/org/apache/poi/ss/formula/functions/MultiOperandNumericFunction.java [hunk 1]
```diff
@@ -20,16 +20,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
-import org.apache.poi.ss.formula.eval.BoolEval;
-import org.apache.poi.ss.formula.eval.ErrorEval;
-import org.apache.poi.ss.formula.eval.EvaluationException;
-import org.apache.poi.ss.formula.eval.NumberEval;
-import org.apache.poi.ss.formula.eval.NumericValueEval;
-import org.apache.poi.ss.formula.eval.OperandResolver;
-import org.apache.poi.ss.formula.eval.RefEval;
-import org.apache.poi.ss.formula.eval.StringValueEval;
-import org.apache.poi.ss.formula.eval.ValueEval;
+import org.apache.poi.ss.formula.eval.*;
```
Comment (centic9) [inline src/java/org/apache/poi/ss/formula/functions/MultiOperandNumericFunction.java]: Please don't revert imports to star-import, best if you adjust your IDE accordingly.

**Review by centic9 — COMMENTED** [review]
CI build is running at https://builds.apache.org/view/P/view/POI/job/POI-Github-PullRequests/65/

File: src/java/org/apache/poi/ss/formula/functions/AggregateFunction.java [hunk 2]
```diff
@@ -257,4 +258,16 @@ protected double evaluate(double[] values) throws EvaluationException {
             return StatsLib.varp(values);
         }
     };
+    public static final Function GEOMEAN = new AggregateFunction() {
+        @Override
+        protected double evaluate(double[] values) throws EvaluationException {
+        	// The library implementation returns 0 for an input sequence like [1, 0]. So this check is necessary.
+        	for (double value: values) {
+        		if (value <= 0) {
+        			throw new EvaluationException(ErrorEval.NUM_ERROR);
```
Comment (pjfanning) [inline src/java/org/apache/poi/ss/formula/functions/AggregateFunction.java]: could you fix the indents in this file to be spaces  (not tabs)?

File: src/java/org/apache/poi/ss/formula/functions/MultiOperandNumericFunction.java (hunk 1 above)
Comment (gallonfizik) [reply src/java/org/apache/poi/ss/formula/functions/MultiOperandNumericFunction.java]: fixed

File: src/java/org/apache/poi/ss/formula/functions/AggregateFunction.java (hunk 2 above)
Comment (gallonfizik) [reply src/java/org/apache/poi/ss/formula/functions/AggregateFunction.java]: fixed

**Comment by pjfanning** [comment]
merged using https://svn.apache.org/viewvc?view=revision&revision=1849042

**Pull request closed without merging**